modules/dists/StencilDist.chpl

```diff
@@ -2346,6 +2352,37 @@ proc StencilDom.dsiLocalSubdomain(loc: locale) {
     return d;
   }
 }
+iter StencilDom.dsiLocalSubdomains(loc: locale) {
+  for locid in dist.chpl__locToLocIdxs(loc) {
+    var inds = chpl__computeBlock(locid, dist.targetLocDom, dist.boundingBox, dist.boundingBox.dims());
+    yield whole[(...inds)];
+  }
+}
+iter StencilArr.dsiLocalSubdomains(loc: locale) {
+  for subdoms in dom.dsiLocalSubdomains(loc) {
+    yield subdoms;
+  }
+}
+
+// These are similar to the normal dsiLocalSubdomains() itertors, but
+// they yield the local blocks as extended by the fluff for the sake
+// of locality-checking rather than just the local blocks.
+//
+iter StencilDom.doiLocalStoredSubdomains() {
+  for locid in dist.chpl__locToLocIdxs(here) {
+
+    // Could calculate the indices directly, as in the normal
+    // subdomains case above, but that's more complicated for
+    // StencilDist, and this isn't on performance-critical paths
+    // anyway, so I'm just looking it up.
```

Comment:
Ah, there's one other piece of history here that I was forgetting, but that this comment reminds me of.  I originally wrote these iterators to take a `locale` argument, and didn't want to do communication to answer the question.  But then in my final review, I realized that the locale argument was never used / always defaulted to `here`, so removed that logic.  So again, I think you're right.

Reply:
I've now updated the comment to indicate that this is guaranteed to be local since we're only ever making this query for `here` (whereas the traditional localSubdomains() queries can be called with any locale argument to determine what subdomains are local to a given locale).